BMS Workspace-git/bms-1227-fw/Phantom_drivers/rtos/source/task_command_line.c

```diff
@@ -0,0 +1,43 @@
+/*
+ * task_command_line.c
+ *
+ *  Created on: Oct 25, 2022
+ *      Author: lonam
+ */
+#include "task_command_line.h"
+
+#include "FreeRTOS.h"
+#include "FreeRTOSConfig.h"
+#include "os_task.h"
+
+#include "Phantom_sci.h"
+#include "hwConfig.h"
+
+static uint8 previous;
+static xTaskHandle commandLineTaskHandle;
+static void CommandLineTask(void* params){
+    while(1){
+        uint8 msg;
+        sciReceive(PC_UART, 1, &msg);
+        if(msg==previous){
+            // do nothing
+        } else if(msg=='1'){
+            UARTprintf("Message 1 received\r\n");
+        } else if(msg=='2'){
+            UARTprintf("Message 2 received\r\n");
+        } else if(msg=='3'){
+            UARTprintf("Message 3 received\r\n");
```

Comment:
Now that the code is written it's easier to see that a switch statement fits more nicely with this logic.

```
switch(param)
{
    case param:
          
           // do the thing 
           break;
}